### PR TITLE
RD-874 Add cluster upgrade system tests

### PR DESCRIPTION
**Pull request by YoniItzhak**
This PR currently takes care only of the 3 nodes case. Once it is approved, I'll continue for the 9 nodes.

### CODE REVIEW

File: cosmo_tester/framework/util.py [hunk 1]
```diff
@@ -665,10 +665,15 @@ def validate_cluster_status_and_agents(manager,
                                        tenant,
                                        expected_brokers_status='OK',
                                        expected_managers_status='OK',
-                                       agent_validation_manager=None):
+                                       agent_validation_manager=None,
+                                       logger=None):
```
Comment (geokala) [inline cosmo_tester/framework/util.py:669]: I think the logger should be mandatory- it makes it more clear where failures occur when you have preceding lines saying what was being attempted.

File: cosmo_tester/test_suites/cluster/conftest.py [hunk 2]
```diff
@@ -177,6 +188,13 @@ def _get_hosts(ssh_key, module_tmpdir, test_config, logger, request,
         for node in hosts.instances:
             node.verify_services_are_running = skip
 
+        if installer_image_name:
+            # The cluster images distro is rhel by default, so let's keep it
```
Comment (geokala) [inline cosmo_tester/test_suites/cluster/conftest.py:192]: It's centos if you specify to use centos in the test config, so we should support that.

File: cosmo_tester/test_suites/cluster/full_cluster_test.py [hunk 3]
```diff
@@ -402,6 +404,59 @@ def test_three_nodes_cluster_teardown(three_nodes_cluster, ssh_key,
     _assert_cluster_status(node1.client)
 
 
+def test_three_nodes_cluster_upgrade(three_nodes_5_1_0_cluster, ssh_key,
+                                     test_config, logger):
+    node1, node2, node3 = three_nodes_5_1_0_cluster
+    nodes_list = [node1, node2, node3]
+    logger.info('Installing example deployment')
+    example = get_example_deployment(node1, ssh_key, logger,
+                                     'cluster_teardown', test_config)
```
Comment (geokala) [inline cosmo_tester/test_suites/cluster/full_cluster_test.py:413]: I'm not convinced 'cluster_teardown' is the right name for this?

Comment (YoniItzhak) [reply cosmo_tester/test_suites/cluster/full_cluster_test.py:413]: You're absolutely right, I missed it.

File: cosmo_tester/test_suites/cluster/conftest.py (hunk 2 above)
Comment (YoniItzhak) [reply cosmo_tester/test_suites/cluster/conftest.py:192]: Thanks, will add a centos support.

File: cosmo_tester/framework/util.py (hunk 1 above)
Comment (YoniItzhak) [reply cosmo_tester/framework/util.py:669]: Agreed, I'll add `logger` to all places that use the function.

**Review by geokala — APPROVED** [review]

**Pull request merged**